### ISSUE TITLE
Crash in Unity on NSNotificationCenter events

### ISSUE THREAD

**Issue by miv** [bugsnag/bugsnag-unity#6]
I've started to get random crashed in iOS build in Unity plugin. Debugging lead me to `BugsnagIosNotifier`. In some cases `NSNotificationCenter` throws unrecognized selector. It happens with every handler, but i've received it most frequently on `lowMemoryWarning` and `orientationChanged`. This was happening sometimes in runtime, but almost every time on resume-after-pause.

I do have another subscriber to these events, but i don't think this can be a problem. After i commented out all subscriptions to `NSNotificationCenter` in `BugsnagIosNotifier,` everything started to work fine.

**Comment by ConradIrwin** [comment]
Hey @miv,

This is super-strange, as we haven't seen any problems with the bugsnag-cocoa notifier doing this before.

Do you have any more details that could help me reproduce this?

**Comment by miv** [comment]
I will investigate a more clear reason after the weekend. For now i can show my logs:

```
-> applicationDidEnterBackground()

-> applicationWillEnterForeground()

2015-01-23 21:15:01.218 myapp[24781:5408645] -[BugsnagSink orientationChanged:]: unrecognized selector sent to instance 0x18c23e70

2015-01-23 21:15:01.225 myapp[24781:5408645] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[BugsnagSink orientationChanged:]: unrecognized selector sent to instance 0x18c23e70'

*** First throw call stack:

(0x2b5cd49f 0x38d83c8b 0x2b5d28b9 0x2b5d07d7 0x2b502058 0x2b585c61 0x2b4e16d5 0x2c213189 0x2ea90a13 0x2ecfc129 0x2ecfb6db 0x2ecfb65d 0x2ecf0943 0x31d4f1e1 0x31d5e0d1 0x2b593d7d 0x2b593041 0x2b591b7b 0x2b4df3c1 0x2b4df1d3 0x328dd0a9 0x2eaeefa1 0x91cc 0x39303aaf)

libc++abi.dylib: terminating with uncaught exception of type NSException

--- next log ---

(Filename: /Applications/buildAgent/work/d63dfc6385190b60/artifacts/iPhonePlayer-armv7Generated/UnityEngineDebug.cpp Line: 56)



2015-01-23 21:06:33.473 myapp[24750:5405808] Received memory warning.

WARNING -> applicationDidReceiveMemoryWarning()

2015-01-23 21:06:33.474 myapp[24750:5405808]  -> myAppController/applicationDidReceiveMemoryWarning

2015-01-23 21:06:33.479 myapp[24750:5405808] -[BugsnagSink lowMemoryWarning:]: unrecognized selector sent to instance 0x1acad820

2015-01-23 21:06:33.489 myapp[24750:5405808] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[BugsnagSink lowMemoryWarning:]: unrecognized selector sent to instance 0x1acad820'

*** First throw call stack:

(0x2b5cd49f 0x38d83c8b 0x2b5d28b9 0x2b5d07d7 0x2b502058 0x2b585c61 0x2b4e16d5 0x2c213189 0x2c217cef 0x2ecf7053 0x487a9c7 0x488c17b 0x487c575 0x487e21f 0x2b5933b1 0x2b591ab1 0x2b4df3c1 0x2b4df1d3 0x328dd0a9 0x2eaeefa1 0x190ec 0x39303aaf)

libc++abi.dylib: terminating with uncaught exception of type NSException


```

**Comment by ehudros** [comment]
This is happening for us as well - getting unknown selector on orientationChanged, probably on resume.
Any workarounds?

**Comment by ConradIrwin** [comment]
Unless I'm going crazy, there's no way this can happen from looking at [the code](https://github.com/bugsnag/bugsnag-cocoa/blob/master/Source/Bugsnag/BugsnagIosNotifier.m#L40-L54). We definitely register an instance of `BugsnagIosNotifier`, not an instance of `BugsnagSink`.

Can either of you share the steps to reproduce this?

**Comment by snmaynard** [comment]
We've released [2.2.4](https://github.com/bugsnag/bugsnag-unity/releases/tag/v2.2.4) that contains a fix for this. Please let me know if you are still having issues with this version!
